ecl/hql/hqlutil.cpp

```diff
@@ -4608,6 +4608,67 @@ IHqlExpression * combineIfsToMap(IHqlExpression * expr)
     return createWrapper(no_map, expr->queryType(), args);
 }
 
+class CaseMatcher
+{
+public:
+    bool matchIfs(IHqlExpression * expr)
+    {
+        if (expr->getOperator() == no_if)
+        {
+            IHqlExpression * test = expr->queryChild(0);
+            if (test->getOperator() == no_eq)
+            {
+                IHqlExpression * search = test->queryChild(0);
+                if (!cond || (search == cond))
+                {
+                    cond = search;
+                    keys.append(*LINK(test->queryChild(1)));
+                    values.append(*LINK(expr->queryChild(1)));
+                    IHqlExpression * elseExpr = expr->queryChild(2);
+                    if (elseExpr)
+                        return matchIfs(elseExpr);
+                    defaultExpr.setown(createNullExpr(expr));
+                    return true;
+                }
+            }
+        }
+        defaultExpr.set(expr);
+        return true;
+    }
```

Comment:
You're correct.  I started off thinking that it would return whether or not ifs had been matched, but when writing it realised it worked better as a recursive function - because it was sensible to translate any leading ifs, even if following ifs didn't match.  That means it only really needs to check worthMapping().
I will change the return type to void to avoid confusion.